### PR TITLE
Chain Upgrade Test

### PR DESCRIPTION
**Pull request by hannydevelop**
Closes: #173 

## Description
Support testing chain upgrades, potentially by integrating with or using ideas from [ibctest](https://github.com/strangelove-ventures/ibctest). Here's a [link to more info](https://docs.google.com/document/d/13OwES9P8tbPz0x76aXpChql-hCRerAoatirgM6UAmas/edit) on what this test entails.

### CODE REVIEW

**Review by cbrit — COMMENTED** [review]
A few items for consideration!

File: Makefile [hunk 1]
```diff
@@ -398,3 +398,58 @@ fail:
...
+
+ORCHESTRATOR_IMAGE := "ghcr.io/peggyjv/gravity-bridge-orchestrator:v0.3.8"
+SOMMELIER_IMAGE := "ghcr.io/peggyjv/sommelier-sommelier:v3.1.0"
+ETHEREUM_IMAGE := "ghcr.io/peggyjv/sommelier-hardhat:v3.1.0"
+SOMM_IMAGE := hannydevelop/sommelier:4.0.1
+
+e2e_build_upgrade_images: e2e_clean_slate
+	@docker pull $(ORCHESTRATOR_IMAGE)
+	@docker tag $(ORCHESTRATOR_IMAGE) orchestrator:3.1.0
+	@docker build -t sommelier:4.0.1 -f Dockerfile .
```
[the remark this answers is not in the document]
Comment (hannydevelop) [reply Makefile:415]: Thanks for catching that, line 410 isn't necessary, I'm removing that in the next commit. As you can see, it's not used here, or anywhere.

```
@docker pull $(ORCHESTRATOR_IMAGE)
	@docker tag $(ORCHESTRATOR_IMAGE) orchestrator:3.1.0
	@docker build -t sommelier:4.0.1 -f Dockerfile .
	@docker pull $(ETHEREUM_IMAGE)
	@docker tag $(ETHEREUM_IMAGE) ethereum:3.1.0
	@docker pull $(SOMMELIER_IMAGE)
	@docker tag $(SOMMELIER_IMAGE) sommelier:3.1.0
```

File: upgrade_tests/chain_upgrade_test.go [hunk 2]
```diff
@@ -0,0 +1,183 @@
...
+			reciever, err := sdk.AccAddressFromBech32("somm1hqf42j6zxfnth4xpdse05wpnjjrgc864vwujxx")
+			s.Require().NoError(err)
+			sendCoin := bank.NewMsgSend(sender, reciever, sdk.Coins{
+				{
+					Denom:  testDenom,
+					Amount: sdk.NewInt(100),
+				},
+			})
+			res, err := s.chain.sendMsgs(*clientCtxs, sendCoin)
+			s.T().Logf("Send coin response:%s", res)
```
[the remark this answers is not in the document]
Comment (hannydevelop) [reply upgrade_tests/chain_upgrade_test.go:31]: Sure, was thinking about that too. Will do that

File: Makefile [hunk 3]
```diff
@@ -398,3 +398,58 @@ fail:
...
+	@docker network prune --force 1>/dev/null 2>/dev/null || true
+	@cd upgrade_tests && go test -c
+
+e2e_test_chain: e2e_clean_upgrade_slate
+	@upgrade_tests/upgrade_tests.test -test.run TestChain -test.failfast -test.v || make -s fail
+
+e2e_upgrade: e2e_clean_upgrade_slate
+	@E2E_SKIP_CLEANUP=true upgrade_tests/upgrade_tests.test -test.failfast -test.v -test.run UpgradeTestSuite -testify.m TestSommChainUpgrade || make -s fail
+
+fail:
```
[the remark this answers is not in the document]
Comment (hannydevelop) [reply Makefile:444]: Good catch!!

File: upgrade_tests/keys.go [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+import (
+	"github.com/cosmos/cosmos-sdk/crypto/hd"
+	"github.com/cosmos/cosmos-sdk/crypto/keyring"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/go-bip39"
+	"github.com/ethereum/go-ethereum/common/hexutil"
+	hdwallet "github.com/miguelmota/go-ethereum-hdwallet"
+)
+
+const DerivationPath = "m/44'/60'/0'/0/0"
```
[the remark this answers is not in the document]
Comment (hannydevelop) [reply upgrade_tests/keys.go:12]: Hmm, I think for better consistency in somm's integration file as well as gravity-bridge, I'll leave this out for now. Thanks for reviewing though, I appreciate

**Pull request closed without merging**